components/suggest/src/rs.rs

```diff
@@ -100,6 +113,41 @@ pub(crate) enum SuggestRecord {
     GlobalConfig(DownloadedGlobalConfig),
 }
 
+/// Enum for the different record types that can be consumed.
+/// Extracting this from the serialization enum so that we can
+/// extend it to get type metadata.
```

Comment:
What do you think about implementing `From<SuggestRecord> for SuggestRecordType`?  I say that because it explicitly shows the relationship between the two types (I think `SuggestRecordType` is just `SuggestRecord` without the field data?  If I saw the mapping I would know for sure).

Reply:
Ah, I didn't know you can do that! I think adding the mapping will definitely show the relationship better, so I'll add that. 👍

Reply:
FYI, [`strum`](https://docs.rs/strum_macros/latest/strum_macros/derive.EnumDiscriminants.html) is made exactly for use cases like this: you want an enum w/o all the data fields. Doing this manually could be tedious as now we will have to add it to various places when adding a new record type. The following will generate an enum `SuggestRecordType` for you automatically with all the bells and whistles. 

```rs
#[derive(Clone, Debug, Deserialize, EnumDiscriminants]
#[strum_discriminants(name(SuggestRecordType))]
pub(crate) enum SuggestRecord {
  #[serde(rename = "icon")]
  #[strum_discriminants(strum(message = "icon"))]  // This maps to what you define in `get_rs_type()` and can be used as `SuggestRecordType::Icon.get_message()`
  Icon,
}
```

Reply:
>   #[strum_discriminants(strum(message = "icon"))]  // This maps to what you define in `get_rs_type()` and can be used as `SuggestRecordType::Icon.get_message()`

I'm a bit confused on how to use this (Rust enums confuses me :p). So, let's say I want to access the "icon" value from `SuggestRecord::Icon`, how do I do that without the match in `get_rs_type` method?

I did manage to reduce duplication on the `SuggestRecordType` though! Thanks for the suggestion!

Reply:
Oh, actually, I think we can use `Strum::Display` to further simplify it:

```rs
use strum_macros::Display as StrumDisplay;

#[derive(Clone, Debug, Deserialize, EnumDiscriminants, StrumDisplay)]
#[strum_discriminants(name(SuggestRecordType))]
#[strum_discriminants(derive(StrumDisplay))]
#[serde(tag = "type")]
pub(crate) enum SuggestRecord {
    #[serde(rename = "icon")]
    #[strum(to_string = "icon")]
    Icon,
    #[serde(rename = "data")]
    #[strum(to_string = "data")]
    AmpWikipedia,
    #[serde(rename = "amo-suggestions")]
    #[strum(to_string = "amo-suggestions")]
    Amo,
    #[serde(rename = "pocket-suggestions")]
    #[strum(to_string = "pocket-suggestions")]
    Pocket,
    #[serde(rename = "yelp-suggestions")]
    #[strum(to_string = "yelp-suggestions")]
    Yelp,
    #[serde(rename = "mdn-suggestions")]
    #[strum(to_string = "mdn-suggestions")]
    Mdn,
    #[serde(rename = "weather")]
    #[strum(to_string = "weather")]
    Weather(DownloadedWeatherData),
    #[serde(rename = "configuration")]
    #[strum(to_string = "configuration")]
    GlobalConfig(DownloadedGlobalConfig),
}

impl SuggestRecordType {
    /// Return the meta key for the last ingested record.
    pub fn last_ingest_meta_key(&self) -> String {
        format!("last_quicksuggest_ingest_{}", self) // because `SuggestRecordType` now implements `Display`
    }
}

// For other places where you need `ingest_record_type.get_rs_type()`, you can just do `ingest_record_type.to_string()`  

```

Reply:
Cool nice! I'll try this as it'll reduce duplication!

Reply:
There is more 😃 , if you stack `#[strum_discriminants(derive(EnumIter))]` to SuggestRecord, then we can also replace `DEFAULT_RECORDS_TYPES` with `SuggestRecordType::iter().collect::<Vec<_>>()`.